lib/events/sessionlog.go

```diff
@@ -49,10 +52,58 @@ type SessionLogger interface {
 	WriteChunk(chunk *SessionChunk) (written int, err error)
 }
 
-// diskSessionLogger implements a disk based session logger. The imporant
+// DiskSessionLoggerConfig sets up parameters for disk session logger
+// associated with the session ID
+type DiskSessionLoggerConfig struct {
+	// SessionID is the session id of the logger
+	SessionID session.ID
+	// EventsFileName is the events file name
+	EventsFileName string
+	// StreamFileName is the byte stream file name
+	StreamFileName string
+	// Clock is the clock replacement
+	Clock clockwork.Clock
+}
+
+// NewDiskSessionLogger creates new disk based session logger
+func NewDiskSessionLogger(cfg DiskSessionLoggerConfig) (*DiskSessionLogger, error) {
+	writtenBytes, err := ReadWrittenBytes(cfg.EventsFileName)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	// create a new session stream file:
+	fstream, err := os.OpenFile(cfg.StreamFileName, os.O_WRONLY|os.O_CREATE|os.O_APPEND, 0640)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	// create a new session file:
+	fevents, err := os.OpenFile(cfg.EventsFileName, os.O_WRONLY|os.O_CREATE|os.O_APPEND, 0640)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	sessionLogger := &DiskSessionLogger{
+		Entry: log.WithFields(log.Fields{
+			trace.Component: teleport.ComponentAuditLog,
+			trace.ComponentFields: log.Fields{
+				"sid": cfg.SessionID,
+			},
+		}),
+		writtenBytes: writtenBytes,
+		sid:          cfg.SessionID,
+		streamFile:   fstream,
+		eventsFile:   fevents,
+		clock:        cfg.Clock,
+		createdTime:  cfg.Clock.Now().In(time.UTC).Round(time.Second),
+	}
+	return sessionLogger, nil
+}
+
+// DiskSessionLogger implements a disk based session logger. The imporant
 // property of the disk based logger is that it never fails and can be used as
 // a fallback implementation behind more sophisticated loggers.
-type diskSessionLogger struct {
+type DiskSessionLogger struct {
```

Comment:
Why is this exported now?

Reply:
I thought it might be useful as a standalone tool (used in my local tests)